### PR TITLE
Reduce ocean output fields

### PR DESCRIPTION
**Pull request by mark-petersen**
This PR reduces the number of variables written in the time-averaged monthly output. It is the result of discussions with the ocean modeling team, as documented on this [confluence page](https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4011983017/Final+ocean+variable+list+for+v3.0). 

This removes the 3D fields zMid and gmKappaScaling, as well as several 2D fields. We decided to turn off the eddy product variables analysis member by default, which saves a large number of 3D fields. These are all variables with 'Times' and 'Squared' in the name, which is up to 20 3D fields, depending on whether GM and the mesoscale eddy parameterization are on. Eddy product variables are typically used at high resolution to compute eddy kinetic energy and similar fields. They are not typically needed at lower resolutions or during spin-up.

[NML]
[BFB]

### CODE REVIEW

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml [hunk 1]
```diff
@@ -1007,7 +1007,7 @@
 <config_AM_transectTransport_transect_group>'all'</config_AM_transectTransport_transect_group>
 
 <!-- AM_eddyProductVariables -->
-<config_AM_eddyProductVariables_enable>.true.</config_AM_eddyProductVariables_enable>
+<config_AM_eddyProductVariables_enable>.false.</config_AM_eddyProductVariables_enable>
```
Comment (xylar) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:1010]: I think this is okay for now but we might want it off for some meshes and on of others in the future.

**Review by xylar — APPROVED** [review]
This looks like what we've been able to agree to.  Thanks @mark-petersen!

File: components/mpas-ocean/cime_config/buildnml [hunk 2]
```diff
@@ -1222,7 +1219,6 @@ def buildnml(case, caseroot, compname):
             lines.append('    <var name="iceRunoffFlux"/>')
             lines.append('    <var name="rainFlux"/>')
             lines.append('    <var name="snowFlux"/>')
-            lines.append('    <var name="vertNonLocalFlux"/>')
```
[the remark this answers is not in the document]
Comment (milenaveneziani) [reply components/mpas-ocean/cime_config/buildnml:1225]: yes, I am also not sure how we would use that variable, to be honest. Just making sure.

[the remark this answers is not in the document]
Comment (vanroekel) [reply components/mpas-ocean/cime_config/buildnml:1225]: `vertNonLocalFlux` is easily recalculable from the surface flux field and the boundary layer depth, no need to keep this.  It is only for simulations where we want to reconstruct turbulent fluxes of heat and salinity, so definitely specialized.

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml (hunk 1 above)
Comment (vanroekel) [reply components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:1010]: I think we can put this on the back burner until we develop the RRS6to18 compset for v3, then we'll want to have the mesh specific enabling.

**Comment by mark-petersen** [comment]
Ran a sanity test, passes `SMS_D.T62_oQU120_ais20.MPAS_LISIO_TEST.chrysalis_gnu`. Also compiled and launched `PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.chrysalis_intel`. It's still in the queue, but produced the namelist and streams options that I expected.

**Comment by mark-petersen** [comment]
Please wait on this PR. I ran a two month test and these variables are still in the time monthly output. I think it is because they were mistakenly put in two packages.
<details>
<summary> details: </summary> <p>

```
pwd
/lcrc/group/e3sm/ac.mpetersen/scratch/chrys/SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_gnu.20231204_152443_hge2j2/run
(dev_compass_1.2.0-alpha.6) chr:run$ ncdump -h SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_gnu.20231204_152443_hge2j2.mpaso.hist.am.timeSeriesStatsMonthly.0001-02-01.nc |grep -i float|grep Times
	float timeMonthly_avg_velocityZonalTimesTemperature_MLE(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityMeridionalTimesTemperature_MLE(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityZonalTimesSalinity_MLE(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityMeridionalTimesSalinity_MLE(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityZonalTimesTemperature_GM(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityMeridionalTimesTemperature_GM(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_normalGMBolusVelocityTimesTemperature(Time, nEdges, nVertLevels) ;
	float timeMonthly_avg_velocityZonalTimesSalinity_GM(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityMeridionalTimesSalinity_GM(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_normalGMBolusVelocityTimesSalinity(Time, nEdges, nVertLevels) ;
```
</details>

**Comment by mark-petersen** [comment]
I ran these tests. 
```
./create_test SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_gnu -q acme-small --walltime 1:00:00
./create_test SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_intel -q acme-small --walltime 1:00:00
```
I found that I had to remove the second package, as well as remove variables from the stream for all of the undesired variables to be removed from the `timeSeriesStatsMonthly` netcdf file. You can see the results of my test here:
```
pwd
/lcrc/group/e3sm/ac.mpetersen/scratch/chrys/SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_gnu.20231204_160734_by1l3f/run
$ f=SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_gnu.20231204_160734_by1l3f.mpaso.hist.am.timeSeriesStatsMonthly.0001-02-01.nc
$ ncdump -h $f |grep Times |grep nCells
$ ncdump -h $f |grep Squa
```

This is now fully tested.

**Comment by mark-petersen** [comment]
We now have a total of 40 3D fields for low resolution runs, as follows:
<details>
<summary> details: </summary> <p>

```
pwd
/lcrc/group/e3sm/ac.mpetersen/scratch/chrys/SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_intel.20231204_164954_6gah5l/run
(dev_compass_1.2.0-alpha.6) chr:run$ f=SMS_Ld62.T62_oQU240.GMPAS-IAF.chrysalis_intel.20231204_164954_6gah5l.mpaso.hist.am.timeSeriesStatsMonthly.0001-02-01.nc
(dev_compass_1.2.0-alpha.6) chr:run$ ncdump -h $f |grep 'nCells, nVertLevels'|wc
     35     175    3770
(dev_compass_1.2.0-alpha.6) chr:run$ ncdump -h $f |grep 'nCells, nVertLevels'
	float timeMonthly_avg_velocityMeridional(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_velocityZonal(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_layerThickness(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_density(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_potentialDensity(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_frazilLayerThicknessTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_vertVelocityTop(Time, nCells, nVertLevelsP1) ;
	float timeMonthly_avg_vertMLEBolusVelocityTop(Time, nCells, nVertLevelsP1) ;
	float timeMonthly_avg_vertGMBolusVelocityTop(Time, nCells, nVertLevelsP1) ;
	float timeMonthly_avg_GMBolusVelocityZonal(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_GMBolusVelocityMeridional(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_frazilTemperatureTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_vertDiffTopOfCell(Time, nCells, nVertLevelsP1) ;
	float timeMonthly_avg_vertViscTopOfCell(Time, nCells, nVertLevelsP1) ;
	float timeMonthly_avg_tendLayerThickness(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_BruntVaisalaFreqTop(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracersTend_temperatureTend(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracersTend_salinityTend(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorizontalAdvectionTendency_temperatureHorizontalAdvectionTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorizontalAdvectionTendency_salinityHorizontalAdvectionTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVerticalAdvectionTendency_temperatureVerticalAdvectionTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVerticalAdvectionTendency_salinityVerticalAdvectionTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVertMixTendency_temperatureVertMixTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVertMixTendency_salinityVertMixTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorMixTendency_temperatureHorMixTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorMixTendency_salinityHorMixTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerSurfaceFluxTendency_temperatureSurfaceFluxTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerSurfaceFluxTendency_salinitySurfaceFluxTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_temperatureShortWaveTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerNonLocalTendency_temperatureNonLocalTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerNonLocalTendency_salinityNonLocalTendency(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVerticalAdvectionTopFlux_temperatureVerticalAdvectionTopFlux(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracerVerticalAdvectionTopFlux_salinityVerticalAdvectionTopFlux(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracers_temperature(Time, nCells, nVertLevels) ;
	float timeMonthly_avg_activeTracers_salinity(Time, nCells, nVertLevels) ;
(dev_compass_1.2.0-alpha.6) chr:run$ ncdump -h $f |grep 'nEdges, nVertLevels'|wc
      5      25     555
(dev_compass_1.2.0-alpha.6) chr:run$ ncdump -h $f |grep 'nEdges, nVertLevels'
	float timeMonthly_avg_normalVelocity(Time, nEdges, nVertLevels) ;
	float timeMonthly_avg_normalMLEvelocity(Time, nEdges, nVertLevels) ;
	float timeMonthly_avg_normalGMBolusVelocity(Time, nEdges, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorizontalAdvectionEdgeFlux_temperatureHorizontalAdvectionEdgeFlux(Time, nEdges, nVertLevels) ;
	float timeMonthly_avg_activeTracerHorizontalAdvectionEdgeFlux_salinityHorizontalAdvectionEdgeFlux(Time, nEdges, nVertLevels) ;
```
</details>

**Comment by mark-petersen** [comment]
Looking at the output fields in the actual tests, I see two we didn’t discuss:
```
float timeMonthly_avg_GMBolusVelocityZonal(Time, nCells, nVertLevels) ;
float timeMonthly_avg_GMBolusVelocityMeridional(Time, nCells, nVertLevels) ;
```
These do not appear in MPAS-Analysis and could be computed with a post-processing step from 
```
float timeMonthly_avg_normalGMBolusVelocity(Time, nEdges, nVertLevels) ;
```
which is recorded. Are there opinions about keeping these? I think it is a convenience versus disk space issue.

**Comment by xylar** [comment]
Thanks for noticing these. They should be removed.

**Review by alicebarthel — APPROVED** [review]
This looks good to me given our discussions. Thanks @mark-petersen

**Comment by jonbob** [comment]
passes:
* PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.chrysalis_intel

with expected NML DIFF. Merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs blessed

**Pull request merged**